primo/utils/override_utils.py

```diff
@@ -0,0 +1,293 @@
+        """
+        violated_operators = {}
+        for operator, groups in self._plugged_list.groupby("Operator Name"):
+            n_wells = len(groups)
+            if n_wells > self.opt_inputs.max_wells_per_owner:
+                violated_operators[operator] = [
+                    n_wells,
+                    groups["API Well Number"].to_list(),
+                ]
+
+        return violated_operators
+
+    def assess_distances(self) -> Dict:
+        """
+        Assess whether the maximum distance between two wells constraint is violated or not
+        """
+        distance_threshold = self.opt_inputs.distance_threshold
+        distance_violations = {}
+        # TODO: Consider calculating these once and for all since back filling
+        # might end up causing multiple iterations here
+        for cluster_id, groups in self._plugged_list.groupby("Project"):
+            well_id = list(groups["API Well Number"])
+            num_well = len(well_id)
+            groups["coords"] = list(zip(groups.Latitude, groups.Longitude))
+            distance_metric_distance = haversine_vector(
+                groups["coords"].to_list(),
+                groups["coords"].to_list(),
+                unit=Unit.MILES,
+                comb=True,
+            )
+            well_distance = {}
+            for well_1 in range(num_well - 1):
+                for well_2 in range(well_1 + 1, num_well):
+                    well_distance = distance_metric_distance[well_1][well_2]
+                    if well_distance > distance_threshold:
+                        distance_violations[
+                            (cluster_id, well_id[well_1], well_id[well_2])
+                        ] = [
+                            cluster_id,
+                            well_id[well_1],
+                            well_id[well_2],
+                            well_distance,
+                        ]
+
+        return distance_violations
+
+    def assess_feasibility(self) -> bool:
+        """
+        Assesses whether current set of selections is feasible
+        """
+        if self.assess_budget() > 0:
+            return False
+
+        if self.assess_dac() > 0:
+            return False
+
+        if self.assess_owner_well_count():
+            return False
+
+        if self.assess_distances():
+            return False
+
+        return True
+
+    def backfill(self) -> List[str]:
+        """
+        If current selections (either due to manual overrides or termination of solver
+        prior to optimality)
+        leave room for budget, the method returns a list of candidates that can
+        be added to the plugging projects without violating any constraints
+
+        Parameters:
+        ----------
+        None
+
+        Returns:
+        --------
+        List of Well API numbers that can be added in the project
+        """
+        additions = []
+        if self.assess_feasibility() is False:
+            # The
```

Comment:
This comment either needs to be finished or removed.

Reply:
ha thanks :-) Fixed!